File: backendm/src/main/java/tn/homrnai/model/Product.java
package tn.homrnai.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;


@Entity
@Data
@Getter
@Setter
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    private double price;
    private String typeP;
    private int quantity;
    @ElementCollection
    private List<String> imageUrls;
    @Temporal(TemporalType.TIMESTAMP)
    private Date date_pub;
    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<DynamicAttribute> attributes = new ArrayList<>();

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;


}

File: marketplaceproject/marketplaceproject/src/app/servicesProduct/Product.ts
import { User } from "../UserService/User";

export interface Product {
  id: number;
  name: string;
  description: string;
  typeP: string;
  price: number;
  quantity: number;
  imageUrls: string[];
  category: any; 
  attributes: { key: string; value: string }[];
  date_pub: Date;
  user:User
  images?: File[]; // For handling file uploads
}
